src/pmdas/bpf/modules/module.h

```diff
@@ -139,4 +141,23 @@ void fill_instids_log2(unsigned int slot_count, pmdaInstid slots[]) {
     }
 }
 
+void fill_instids(unsigned int slot_count, pmdaInstid **slots) {
+    for (int i = 0; i < slot_count; i++) {
+        char *string;
+        pmdaInstid    *slots_tmp;
+        size_t size = (i + 1) * sizeof(pmdaInstid);
+
+        if ((slots_tmp = (pmdaInstid *)realloc(*slots, size)) == NULL)
+            pmNotifyErr(LOG_ERR, "pmdaInstid: realloc err: %d", PM_FATAL_ERR);
+        *slots = slots_tmp;
+
+        int ret = asprintf(&string, "%d", i);
+        if (ret > 0) {
+            (*slots)[i].i_name = (char *) malloc(100 * sizeof(char));
```

Comment:
This line allocates 100 bytes, but in the next line the pointer to these 100 bytes gets overridden.
These 100 bytes in memory will be lost forever, because there is no pointer to this memory block anymore (resulting in a memory leak). Afaics this line should just be deleted, in the next line the `string` variable is assigned to it, which contains memory allocated by the `asprintf` function.

Reply:
Updated.